src/Fable.Compiler/MSBuildCrackerResolver.fs

```diff
@@ -0,0 +1,179 @@
+
+            return output.Trim()
+        }
+        |> Async.AwaitTask
+
+    let private dotnet_msbuild (fsproj: FullPath) (args: string) : Async<string> =
+        dotnet_msbuild_with_defines fsproj args List.empty
+
+    let mkOptionsFromDesignTimeBuildAux (fsproj: FileInfo) (options: CrackerOptions) : Async<ProjectOptionsResponse> =
+        async {
+            let! targetFrameworkJson =
+                let configuration =
+                    if String.IsNullOrWhiteSpace options.Configuration then
+                        ""
+                    else
+                        $"/p:Configuration=%s{options.Configuration} "
+
+                dotnet_msbuild
+                    fsproj.FullName
+                    $"%s{configuration} --getProperty:TargetFrameworks --getProperty:TargetFramework"
+
+            let targetFramework =
+                let properties =
+                    JsonDocument.Parse targetFrameworkJson
+                    |> fun json -> json.RootElement.GetProperty "Properties"
+
+                let tf, tfs =
+                    properties.GetProperty("TargetFramework").GetString(),
+                    properties.GetProperty("TargetFrameworks").GetString()
+
+                if not (String.IsNullOrWhiteSpace tf) then
+                    tf
+                else
+                    tfs.Split ';' |> Array.head
+
+
+            // When CoreCompile does not need a rebuild, MSBuild will skip that target and thus will not populate the FscCommandLineArgs items.
+            // To overcome this we want to force a design time build, using the NonExistentFile property helps prevent a cache hit.
+            let nonExistentFile = Path.Combine("__NonExistentSubDir__", "__NonExistentFile__")
+
+            let properties =
+                [
+                    "/p:Fable=True"
+                    if not (String.IsNullOrWhiteSpace options.Configuration) then
+                        $"/p:Configuration=%s{options.Configuration}"
+                    $"/p:TargetFramework=%s{targetFramework}"
+                    "/p:DesignTimeBuild=True"
+                    "/p:SkipCompilerExecution=True"
+                    "/p:ProvideCommandLineArgs=True"
+                    // See https://github.com/NuGet/Home/issues/13046
+                    "/p:RestoreUseStaticGraphEvaluation=False"
+                    // Avoid restoring with an existing lock file
+                    "/p:RestoreLockedMode=false"
+                    "/p:RestorePackagesWithLockFile=false"
+                    // We trick NuGet into believing there is no lock file create, if it does exist it will try and create it.
+                    " /p:NuGetLockFilePath=Fable.lock"
+                    // Avoid skipping the CoreCompile target via this property.
+                    $"/p:NonExistentFile=\"%s{nonExistentFile}\""
+                ]
+                |> List.filter (String.IsNullOrWhiteSpace >> not)
```

Comment:
To me it looks like this filter always return the full array as we don't have any empty string in it.

We the change of scope of `configuration` we will probably need to keep this line.